app/main/routes_devices.py

```diff
@@ -33,6 +33,19 @@ def handle_devices():
         uid = str(request.form['uid'])
         alias = str(request.form['alias'])
 
+        # uid iand alias are required
+        if len(uid) == 0 or len(alias) == 0:
```

Comment:
Make this if/elif instead of checking 2x.. also fix "iand" in comment?

Reply:
I wasn't a fan of this approach either, but wanted to give a clear message regardless of it being one or the other or both.   I can make it as simple "Machine/Product ID and/or Alias are required" without any additional checks otherwise I am still going to need something like

```
if len(uid) == 0 and len(alias) == 0:
                error = f'Machine/Product ID and Alias are required'
            elif len(uid) == 0:
                error = f'Machine/Product ID is required'
            else:
                error = f'Alias is required'
```
 
Let me know which you prefer

Reply:
BTW, i'd typically do the validation in javascript for a better user experience, just wanted to get this failsafe in since the user cannot recover without editing the config.yaml

Reply:
I'd say for simplicity, just split it if/elif or just do a len check on both (your first if), and then for the error say "Both Machine/Product ID AND Alias are required" - KISS :)